src/fprime/util/cookiecutter_wrapper.py

```diff
@@ -77,23 +55,22 @@ def run_impl(deployment: Path, path: Path, platform: str, verbose: bool):
     hpp_src = hpp_files_template[0]
     cpp_src = cpp_files_template[0]
 
-    # Copy files without headers
-    for src, dest in [(hpp_src, hpp_dest), (cpp_src, cpp_dest)]:
-        with open(src, "r") as file_handle:
-            lines = file_handle.readlines()
-        lines = lines[11:]  # Remove old header
-        with open(dest, "a") as file_handle:
-            file_handle.write("".join(lines))
-
-    removals = [Path(hpp_src), Path(cpp_src)]
-    for removal in removals:
-        os.remove(removal)
+    # Move (and overwrite) files from *.(c|h)pp-template to *.(c|h)pp
+    shutil.move(hpp_src, hpp_dest)
+    shutil.move(cpp_src, cpp_dest)
+
     return True
 
 
-def add_to_cmake(list_file: Path, comp_path: Path):
-    """Adds new component or port to CMakeLists.txt"""
-    print(f"[INFO] Found CMakeLists.txt at '{list_file}'")
+def add_to_cmake(list_file: Path, comp_path: Path, project_root: Path = None):
```

Comment:
This function doesn't look like it will work with arbitrary CMake structure.

e.g. it only works with the parent cmake list, not `project.cmake` nor if there are empty directories in between.

In other implementations of this we have recursed upward to project root looking for:

1. If `project.cmake` then add to it
2. Else if `CMakeLists.txt` then add to it
3. Else, repeat on parent directory

Reply:
Added my take at implementing this, let me know how that looks